stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -5124,16 +5124,27 @@ extension ${Self} {
     }
   }
 
+  /// Gets the native Dictionary hidden inside this NSDictionary
+  /// or returns nil.
   public static func _bridgeFromObjectiveCAdoptingNativeStorageOf(
     _ s: AnyObject
   ) -> ${Self}<${TypeParameters}>? {
-    if let bridgingStorage =
-        s as AnyObject as? _SwiftDeferredNS${Self}<${TypeParameters}> {
-      // If `NS${Self}` is actually native storage of `${Self}` with key
-      // and value types that the requested ones match exactly, then just
-      // re-wrap the native storage.
-      return ${Self}<${TypeParameters}>(_nativeStorage: bridgingStorage.nativeStorage)
+
+    // Try all three NSSelf impls that we currently provide.
```

Comment:
Would it be better to cast to a common base class and if it succeeds, unconditionally call a method that does the rest of the work?  Just askin'.